presto-main/src/main/java/io/prestosql/operator/window/FirstValueFunction.java

```diff
@@ -27,11 +27,21 @@
 {
     private final int argumentChannel;
```

Comment:
It would be nice to have some benchmark that measure the speedup? Did you do some performance testing?

Reply:
I did some ad-hoc testing to show that it only checks the new values in each window, where the previous implementation rechecks the window for each row. I didn't do anything at scale personally, although there have been some significant performance hits observed with real data. What would you suggest?

Reply:
There is io.prestosql.operator.BenchmarkWindowOperator that benchmarks `ROW_NUMBER` function. Based on that we could write benchmarks for functions that you are improving.

WDYT?